discovery/gossiper_test.go

```diff
@@ -3545,3 +3550,41 @@ func assertMessage(t *testing.T, expected, got lnwire.Message) {
 			spew.Sdump(got))
 	}
 }
+
+// TestSplitAnnouncementsCorrectSubBatches checks that we split a given
+// sizes of announcement list into the correct number of batches
+func TestSplitAnnouncementsCorrectSubBatches(t *testing.T) {
+	t.Parallel()
+
+	const subBatchSize = 10
+	const minimumSubBatchSize = 1
+
+	announcementBatchSizes := []int{2, 5, 20, 45, 80, 100, 1005}
+	expectedNumberMiniBatches := []int{1, 1, 2, 5, 8, 10, 101}
+
+	lengthAnnouncementBatchSizes := len(announcementBatchSizes)
+	lengthExpectedNumberMiniBatches := len(expectedNumberMiniBatches)
+
+	assert.Equal(t,
```

Comment:
fwiw, we typically would write these assertions as:
```golang
if lengthAnnouncementBatchSizes != lengthAnnouncementBatchSizes {
        t.Fatalf(...)
}
```
to avoid importing another external package. See the other test cases for more examples